src/openai/lib/azure.py

```diff
@@ -287,8 +287,7 @@ def _prepare_options(self, options: FinalRequestOptions) -> None:
 
         azure_ad_token = self._get_azure_ad_token()
         if azure_ad_token is not None:
-            if headers.get("Authorization") is None:
-                headers["Authorization"] = f"Bearer {azure_ad_token}"
+            headers["Authorization"] = f"Bearer {azure_ad_token}"
```

Comment:
Under what (if any) circumstances would/could the caller provide both an authorization header AND a token provider callback?

Reply:
Never mind - separate check in the _get_azure_ad_token call.

Reply:
I can't think of a scenario where the caller would do that... or what behavior they would expect. Unless the idea is to set some initial value and then allow the token provider to update it, but I still can't wrap my head around where that would make sense.